### PR TITLE
UNT-T30314 - Added new animation reaction view

### PR DESCRIPTION
**Pull request by faaizdaglawala**

### CODE REVIEW

File: SSSwiftUIAnimations/Examples/ExampleReactionAnimationView.swift [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import SwiftUI
+
+struct ExampleReactionAnimationView: View {
+    
+    // MARK: - Variables
+    
+    var body: some View {
+        VStack {
+            SSReactionAnimationView(
+                viewModel: SSReactionAnimationViewModel(
```
Comment (rahuly-simform) [inline SSSwiftUIAnimations/Examples/ExampleReactionAnimationView.swift:17]: Can we make a StyleClass for providing the Styling Attributes? We can keep the Viewmodel for Logical part of animation.

File: SSSwiftUIAnimations/Sources/ReactionAnimation/ReactionAnimationViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+//
+//  Created by Faaiz Daglawala on 15/12/22.
+//
+
+import Foundation
+//import Observation
+import SwiftUI
+
+/// ViewModel for managing the reaction animation in SwiftUI.
+class SSReactionAnimationViewModel: ObservableObject {
```
Comment (rahuly-simform) [inline SSSwiftUIAnimations/Sources/ReactionAnimation/ReactionAnimationViewModel.swift:13]: We can mark this as public if we are using it outside the package.

File: SSSwiftUIAnimations/Sources/ReactionAnimation/ReactionAnimationViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+    @Published private(set) var animationProgress: Double = 0
+    @Published private(set) var isSelected: Bool = false
+
+    // MARK: - Initializers
+    
+    /// Default initializer
+    init() { }
+    
+    /// Custom initializer to set colors for outer and inner circles, and heart icons.
+    init(outerCircleColor: Color,
```
Comment (rahuly-simform) [inline SSSwiftUIAnimations/Sources/ReactionAnimation/ReactionAnimationViewModel.swift:82]: We can put a documented comment over the init.

File: SSSwiftUIAnimations/Sources/ReactionAnimation/ReactionAnimationView.swift [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+        
+        /// Binding to the parent view's ViewModel to access animation properties.
+        @ObservedObject var viewModel: SSReactionAnimationViewModel
+        
+        // MARK: - Body
+        
+        var body: some View {
+            ZStack {
+                // Seven Large Circles around the central circle, each positioned based on angle calculation.
+                ForEach(0..<7) { index in
```
Comment (rahuly-simform) [inline SSSwiftUIAnimations/Sources/ReactionAnimation/ReactionAnimationView.swift:98]: Can we have dynamic number for bubble?

File: SSSwiftUIAnimations/Sources/ReactionAnimation/ReactionAnimationViewModel.swift (hunk 2 above)
Comment (faaizdaglawala) [reply SSSwiftUIAnimations/Sources/ReactionAnimation/ReactionAnimationViewModel.swift:13]: We are now not using this outside the package now, as there is a new style struct created which is used outside the package.

**Review by bhargavbajani-simformsolutions — APPROVED** [review]

**Pull request merged**